host/tests/dsCompositeIn_L3_Tests/dsCompositeInHelperClass.py

```diff
@@ -0,0 +1,158 @@
+        """
+        Initializes the test class with test name, setup configuration, and sessions for the device.
+
+        Args:
+            testName (str) : name of the test
+            qcId (str): QC ID of the test.
+            log (class, optional): Parent log class. Defaults to None.
+        """
+        self.testName  = ""
+        self.testSetupPath = os.path.join(dir_path, "dsCompositeIn_L3_testSetup.yml")
+        self.moduleName = "dsCompositeIn"
+        self.rackDevice = "dut"
+
+        super().__init__(testName, qcId, log)
+
+        # Load test setup configuration
+        self.testSetup = ConfigRead(self.testSetupPath, self.moduleName)
+
+        # Open Sessions for hal test
+        self.hal_session = self.dut.getConsoleSession("ssh_hal_test")
+
+         # Create user response Class
+        self.testUserResponse = utUserResponse()
+
+        # Get path to device profile file
+        self.deviceProfile = os.path.join(dir_path, self.cpe.get("test").get("profile"))
+
+        self.deviceDownloadPath = self.cpe.get("target_directory")
+
+    def testDownloadAssets(self):
+        """
+        Downloads the test artifacts and streams listed in the test setup configuration.
+
+        This function retrieves streams and other necessary files and
+        saves them on the DUT (Device Under Test).
+
+        Args:
+            None
+        """
+
+        # List of streams with path
+        self.testStreams = []
+
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+
+        # Download test artifacts to device
+        url = test.get("artifacts")
+        if url is not None:
+            self.downloadToDevice(url, self.deviceDownloadPath, self.rackDevice)
+
+        url = self.testSetup.get("assets").get("device").get(self.testName).get("streams")
+        # Download test streams to device
+        if url is not None:
+            self.downloadToDevice(url, self.deviceDownloadPath, self.rackDevice)
+            for streampath in url:
+                self.testStreams.append(os.path.join(self.deviceDownloadPath, os.path.basename(streampath)))
+
+    def testCleanAssets(self):
+        """
+        Removes the downloaded assets and test streams from the DUT after test execution.
+
+        Args:
+            None
+        """
+        self.deleteFromDevice(self.testStreams)
+
+    def testRunPrerequisites(self):
+        """
+        Executes prerequisite commands listed in the test setup configuration file on the DUT.
+
+        Args:
+            None
+        """
+
+        # Run commands as part of test prerequisites
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+        cmds = test.get("execute")
+        if cmds is not None:
+            for cmd in cmds:
+                self.writeCommands(cmd)
+
+        # Run commands as part of test prerequisites
+        if self.testName == "test3_ScaleAndVerify_Video":

```

Comment:
This prerequisite command are required only in case of scale test case so I have added like only if the test name is "test3_ScaleAneVerify" need to execute that commands in testsession

Reply:
done